polus-training-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,520 @@
+    assert num_images_trained == num_labels_trained, "The number of images does not match the number of ground truths for training"
+    assert num_images_tested == num_labels_tested, "The number of images does not match the number of ground for testing"
+    num_trained = num_images_trained
+    num_tested = num_images_tested
+
+    del num_images_trained
+    del num_images_tested
+    del num_labels_trained
+    del num_labels_tested
+
+    # Need a list of numpy arrays to feed to SplineDist
+    array_images_trained = []
+    array_labels_trained = []
+    array_images_tested = []
+    array_labels_tested = []
+
+    # Neural network parameters
+    axis_norm = (0,1)
+    n_channel = 1 # this is based on the input data
+    
+    model_dir_name = '.'
+    model_dir_path = os.path.join(output_directory, model_dir_name)
+
+    # Get the testing data for neural network
+    logger.info("\n Starting to Load Test Data ...")
+    tenpercent_tested = np.floor(num_tested/10)
+    for i in range(num_tested):
+
+        image = X_val[i]
+        label = Y_val[i]
+        base_image = os.path.basename(str(image))
+        base_label = os.path.basename(str(label))
+        assert base_image == base_label, "{} and {} do not match".format(base_image, base_label)
+
+        # The original image
+        br_image = BioReader(image)
+        im_array = br_image[:,:,0:1,0:1,0:1]
+        im_array = im_array.reshape(br_image.shape[:2])
+        array_images_tested.append(normalize(im_array,pmin=1,pmax=99.8,axis=axis_norm))
+
+        # The corresponding label for the image
+        br_label = BioReader(label)
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_tested.append(fill_label_holes(lab_array))
+        
+        assert im_array.shape == lab_array.shape, "{} and {} do not have matching shapes".format(base_image, base_label)
+
+        if (i%tenpercent_tested == 0) and (i!=0):
+            logger.info("Loaded ~{}% of Test Data".format(np.ceil((i/num_tested)*100), num_tested))
+    logger.info("Done Loading Testing Data")
+
+    # Read the input images used for training
+    logger.info("\n Starting to Load Train Data ...")
+    tenpercent_trained = np.floor(num_trained/10)
+    contoursize_max = 0 # gets updated if models have not been created for config file
+    for i in range(num_trained):
+
+        br_image = BioReader(X_trn[i])
+        br_image_shape = br_image.shape[:2]
+        im_array = br_image[:,:,0:1,0:1,0:1]
+        im_array = im_array.reshape(br_image_shape)
+        array_images_trained.append(normalize(im_array,pmin=1,pmax=99.8,axis=axis_norm))
+
+        if i == 0:
+            n_channel = br_image.shape[2]
+
+        br_label = BioReader(Y_trn[i])
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_trained.append(fill_label_holes(lab_array))
```

Comment:
Same as above. Make sure to close BioReaders.

Also, all images in WIPP should only have 1 channel, so `n_channel` should always be 1. If you want to be able to do multi-channel images, you should try to merge images using a `filepattern` with `c` variable.